### DISCUSSION TITLE
uvicorn.run reload does not use watchfiles

### DISCUSSION THREAD

**Discussion by omerXfaruq** [Kludex/uvicorn#1888] [Potential Issue] [answered]
hello!

I think the default reload process is watchfiles, see [here](https://github.com/encode/uvicorn/blob/master/docs/settings.md#development), however watchfiles is not used when the app is run with uvicorn.run, instead statreload is being used. Also, if I pip install watchgod, uvicorn uses watchgod successfully, however it is a deprecated package, and it is renamed to watchfiles.

> The package under the name watchgod will not be further developed and will only receive critical security fixes.

I have an always running application which does updates using the reload feature of uvicorn. Today I realized that while the application was idle it was always using %40 of a cpu core. I thought it could be because of reload, and my intuition was correct. Watchgod does not consume cpu at all, which is great.

You can see my usage example [here](https://github.com/FarukOzderim/Memory-Vault/blob/48c25e25093aae5c01151d90b9e8c743a18283a4/src/__init__.py#L58).

**Comment by Kludex** [comment, marked as answer]
I just tried here. `watchfiles` is being used on `uvicorn.run(<app_path>, reload=True)`.

**Reply by omerXfaruq** [reply]
Interesting, thx for checking.

**Reply by Kludex** [reply]
Did you find out what the issue was?

**Reply by omerXfaruq** [reply]
No, might be due to specific versions I'm using in that repo, but don't think it is important if it is not happening with you.
